postgres/tests/test_statements.py

```diff
@@ -2086,3 +2091,50 @@ def _run_queries():
 
     for conn in connections.values():
         conn.close()
+
+
+@pytest.mark.unit
+def test_get_query_metrics_payload_rows():
+    config = PostgresConfig({"host": "host", "username": "user"}, {}, None)
+    statement_metrics = PostgresStatementMetrics({}, config, None)
+    wrapper = {}
+
+    TestCase = namedtuple('TestCase', 'rows max_size expected')
+
+    test_cases = [
+        # Empty
+        TestCase(rows=[], max_size=1000, expected=0),
+        # Single row too large
+        TestCase(rows=[{'query': 'a' * 1000}], max_size=10, expected=0),
+        # Both rows fit
+        TestCase(rows=[{'query': 'a' * 20}, {'query': 'b' * 20}], max_size=100, expected=1),
+        # Split rows once
+        TestCase(
+            rows=[{'query': 'a' * 20}, {'query': 'b' * 20}, {'query': 'a' * 20}, {'query': 'b' * 20}],
```

Comment:
why does this case split the rows? isnt 20*4 < 100? or does the payload size calculation account for the size of the tags too (im assuming thats what it is)